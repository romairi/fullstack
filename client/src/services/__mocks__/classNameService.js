module.exports = {
    buildClassName: jest.fn(() => 'COMPUTED_CLASS_NAME'),
};
