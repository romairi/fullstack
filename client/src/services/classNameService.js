export const buildClassName = classes => classes.filter(c => !!c).join(' ');
