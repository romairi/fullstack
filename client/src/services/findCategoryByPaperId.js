export function findCategoryByPaperId(categories, paperId) {
    return categories.find(c => c.paperItems.find(paper => paper.paperId === paperId));
}
