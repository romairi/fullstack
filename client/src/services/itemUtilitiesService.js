export function searchByFields(items, field) {
    return items.filter(item => [item.title, item.summary, ...item.authors.filter(a => Array.isArray(a) && a.length > 0).map(a => a[0])]
        .map(text => text.toLowerCase())
        .filter(text => text.includes(field.toLowerCase()))
        .length > 0
    );
}
