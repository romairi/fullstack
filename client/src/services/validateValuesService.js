export function reformatErrors(details) {
    return details.reduce((acc, cur) => {
        acc[cur.context.key] = cur.message;
        return acc;
    }, {});
}

export function validateValues(values, schema) {
    const results = schema.validate(values, {abortEarly: false});
    let errors = {};

    if (results.error) {
        errors = reformatErrors(results.error.details);
    }
    return errors;
}

