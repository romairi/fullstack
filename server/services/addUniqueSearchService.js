function formatSearchTags(listTags) {
    return listTags.map(tag => tag.toLowerCase()).sort().join();
}

function uniqueSearch(userSearches, includeTags, excludeTags) {
    const includeList = [];
    const excludeList = [];
    let i = 0;

    while (userSearches.length > i) {
        includeList.push(formatSearchTags(userSearches[i].include_tags));
        excludeList.push(formatSearchTags(userSearches[i].exclude_tags));
        i++;
    }

    const searchIncTags = formatSearchTags(includeTags.map(item => item));
    const searchExcTags = formatSearchTags(excludeTags.map(item => item));
    const foundInclude = includeList.find(item => item === searchIncTags);
    const foundExclude = excludeList.find(item => item === searchExcTags);
    const includeTagExists = searchIncTags.includes(foundInclude);
    const excludeTagExists = searchExcTags.includes(foundExclude);

    return (includeTagExists && excludeTagExists);
}

module.exports = {uniqueSearch};
