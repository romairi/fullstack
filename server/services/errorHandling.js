function logErrors(err, req, res, next) {
    console.error(err.stack);
    next(err);
}

function clientErrorHandler(err, req, res, next) {
    if (req.xhr) {
        res.status(500).send({error: 'Something failed!'});
    } else {
        next(err);
    }
}

function errorHandler(err, req, res, next) {
    res.status(500);
    res.send('500 Internal Server Error');
}

module.exports = {logErrors, clientErrorHandler, errorHandler};