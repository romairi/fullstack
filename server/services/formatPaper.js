function formatPaper({id, links, ...paperArgs}) {
    const pdfLinkObject = Array.isArray(links) ? links.find(link => link.title === 'pdf') : null;
    const pdfLink = pdfLinkObject ? pdfLinkObject.href : null;
    return {
        ...paperArgs,
        pdfLink,
        paperId: id,
    };
}

module.exports = {formatPaper};
