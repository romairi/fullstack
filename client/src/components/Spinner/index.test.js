import React from 'react';
import renderer from 'react-test-renderer';
import SpinnerContainer from './index';
import {buildClassName} from "../../services/classNameService";

jest.mock('../../services/classNameService');
jest.mock('@material-ui/core/CircularProgress', () => () => <div className="CircularProgress" />);

describe('Spinner', () => {
    describe('render', () => {
       it('should render the children - isLoading is false', () => {
           const component = renderer.create(
               <SpinnerContainer isLoading={false}>
                   <div>CHILDREN</div>
               </SpinnerContainer>,
           );
           expect(component.toJSON()).toMatchSnapshot();
           expect(buildClassName.mock.calls.pop()).toMatchSnapshot();
       });
       it('should render the spinner - isLoading is true', () => {
           const component = renderer.create(
               <SpinnerContainer isLoading={true}>
                   <div>CHILDREN</div>
               </SpinnerContainer>,
           );
           expect(component.toJSON()).toMatchSnapshot();
           expect(buildClassName.mock.calls.pop()).toMatchSnapshot();
       });
    });
});
