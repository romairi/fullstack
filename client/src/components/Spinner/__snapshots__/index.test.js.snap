// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner render should render the children - isLoading is false 1`] = `
<div
  className="COMPUTED_CLASS_NAME"
>
  <div>
    CHILDREN
  </div>
</div>
`;

exports[`Spinner render should render the children - isLoading is false 2`] = `
Array [
  Array [
    "spinner_container",
    false,
  ],
]
`;

exports[`Spinner render should render the spinner - isLoading is true 1`] = `
<div
  className="COMPUTED_CLASS_NAME"
>
  <div
    className="CircularProgress"
  />
</div>
`;

exports[`Spinner render should render the spinner - isLoading is true 2`] = `
Array [
  Array [
    "spinner_container",
    "loading",
  ],
]
`;
